framework_tvb/tvb/interfaces/web/controllers/flow_controller.py

```diff
@@ -158,7 +158,9 @@ def show_group_of_algorithms(self, step_key, algorithm_ids):
             algorithm_id = int(i)
             algorithm = self.flow_service.get_algorithm_by_identifier(algorithm_id)
             algorithm.link = self.get_url_adapter(step_key, algorithm_id)
-            algorithm.input_tree = self.flow_service.prepare_adapter(algorithm)
+            adapter_form = self.flow_service.prepare_adapter_form(self.flow_service.prepare_adapter(algorithm), project.id)
+            algorithm.displayname = algorithm.displayname + ': ' + adapter_form.get_connectivity_label()
```

Comment:
Why not modify property displayname (or perhaps _ui_name) in the BCT Adapters directly?
I think it will be more uniform